inc/html.class.php

```diff
@@ -3796,19 +3804,35 @@ static function autocompletionTextField(CommonDBTM $item, $field, $options=array
       }
       $params['user']   = -1;
       $params['option'] = '';
+      $params['type']   = 'text';
+      $params['required']  = false;
 
       if (is_array($options) && count($options)) {
          foreach ($options as $key => $val) {
             $params[$key] = $val;
          }
       }
+
+      $rand = (isset($params['rand']) ? $params['rand'] : mt_rand());
+      $name    = "field_".$params['name'].$rand;
+
       $output = '';
       if ($CFG_GLPI["use_ajax_autocompletion"]) {
-         $rand    = mt_rand();
-         $name    = "field_".$params['name'].$rand;
-         $output .=  "<input ".$params['option']." id='text$name' type='text' name='".
+         $output .=  "<input ".$params['option']." id='text$name' type='{$params['type']}' name='".
                        $params['name']."' value=\"".self::cleanInputText($params['value'])."\"
-                       class='autocompletion-text-field'>";
+                       class='autocompletion-text-field'";
+
+         if ($params['required'] == true) {
+            $output .= " required='required'";
+         }
+
+         if (isset($params['attrs'])) {
```

Comment:
attrs is a new options for autocompletiontextfield, no ? if yes, we need a default entry in $params and a dockblock